src/screenshot-window.ts

```diff
@@ -12,16 +12,25 @@ export function createScreenshotWindow( captureUrl: string ) {
 		webPreferences: { session: newSession },
 	} );
 
-	const finishedLoading = new Promise< void >( ( resolve ) => {
-		window.webContents.on( 'did-finish-load', () => resolve() );
-	} );
+	const responseStatusCodePromise = new Promise< void >( ( resolve, reject ) => {
+		newSession.webRequest.onCompleted( ( details ) => {
```

Comment:
Adding a URL filter did not work well in my testing. Likely because of potential redirects.